internal/nfs/controller/volume.go

```diff
@@ -251,18 +254,22 @@ func (nv *NFSVolume) deleteExportCommand(cmd, nfsCluster string) []string {
 // getNFSCluster fetches the NFS-cluster name from the CephFS journal.
 func (nv *NFSVolume) getNFSCluster() (string, error) {
 	if !nv.connected {
-		return "", fmt.Errorf("can not get NFS-cluster for %q: not connected", nv)
+		return "", fmt.Errorf("can not get NFS-cluster for %q: %w", nv, ErrNotConnected)
 	}
 
 	fs := fscore.NewFileSystem(nv.conn)
 	fsName, err := fs.GetFsName(nv.ctx, nv.fscID)
 	if err != nil {
-		return "", fmt.Errorf("failed to get filesystem name for ID %x: %w", nv.fscID, err)
+		if errors.Is(err, util.ErrPoolNotFound) {
+			return "", fmt.Errorf("%w for ID %x: %v", ErrFilesystemNotFound, nv.fscID, err)
+		}
+
+		return "", fmt.Errorf("filesystem name for ID %x %w: %v", nv.fscID, ErrNotFound, err)
```

Comment:
note that `ErrNotFound` will cause the caller to assume that the NFS-volume does not exist (anymore). If there is some other error (permissions?), it may not be correct. It is probably better to not include `ErrNotFound`, in the non-specific error return.